nbconvert/nbconvertapp.py

```diff
@@ -323,19 +338,23 @@ def init_single_notebook_resources(self, notebook_filename):
 
         return resources
 
-    def export_single_notebook(self, notebook_filename, resources):
+    def export_single_notebook(self, shell_input, resources):
```

Comment:
can you rename `shell_input` here back to `notebook_filename` (also in docstring)

You can assign after but the comment than changing the signature break api is right.